classes/onos.py

```diff
@@ -0,0 +1,91 @@
+import os
+import requests
+
+from classes.deploy_target import DeployTarget
+link_lines = ""
+device_lines = ""
+host_lines = ""
+
+class Onos(DeployTarget):
+
+    def __init__(self, base_url, auth=(os.getenv("ONOSUSER"), os.getenv("ONOSPASS"))):
```

Comment:
`credentials` might be a better name for this.
`auth` sounds like an authentication method to me.

```suggestion
    def __init__(self, base_url, credentials=(os.getenv("ONOSUSER"), os.getenv("ONOSPASS"))):
```